app/lib/common/models/medication.dart

```diff
@@ -56,3 +56,8 @@ MedicationWithGuidelines medicationWithGuidelinesFromHTTPResponse(
 ) {
   return MedicationWithGuidelines.fromJson(jsonDecode(resp.body));
 }
+
+Iterable<dynamic> idsFromHTTPResponse(Response resp) {
+  final idsList = jsonDecode(resp.body) as List<dynamic>;
+  return idsList.map((el) => el['id']);
+}
```

Comment:
```suggestion
Iterable<int> idsFromHTTPResponse(Response resp) {
  final idsList = jsonDecode(resp.body) as List<dynamic>;
  return idsList.map((el) => el['id']) as List<int>;
}
```

for more typesafety

Reply:
this is throwing an Exception in l.62: `_CastError (type 'MappedListIterable<dynamic, dynamic>' is not a subtype of type 'List<int>' in type cast)`

Reply:
I'll commit something that works but is four lines longer, maybe you have an idea how to shorten it